src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -88,6 +88,13 @@ export class NotebookLinkHandler {
 			 * We return the absolute path for the link so that it will get used in the as the href for the anchor HTML element
 			 * (in linkCalloutDialog document.execCommand('insertHTML') and therefore will call getLinkURL() with HTMLAnchorElement to then get the relative path
 			*/
+			// Need to encode URI here in order for user to click the proper encoded link in WYSIWYG
+			// skip encoding it if it's already encoded
+			let encodedLinkURL = this._link;
+			if (this._isFile || encodedLinkURL === unescape(encodedLinkURL)) {
```

Comment:
1. Why do you skip this if it's a file?
2. Why are you still using a different unescape function than encode function? And why are you using unescape to begin with? The [docs page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape) has a big warning about not using it at all and specifically calls out not to use it to decode URIs (which this is...)